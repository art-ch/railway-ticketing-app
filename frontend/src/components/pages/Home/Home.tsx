'use client';

import React from 'react';

import { TrainScheduleDisplay } from './components/TrainScheduleDisplay/TrainScheduleDisplay';
import { EmbeddedDisplay } from './components/EmbeddedDisplay/EmbeddedDisplay';
import { Clock } from './components/Clock/Clock';
import { Train } from './components/Train/Train';
import { Train as TrainType } from 'railway-ticketing-app-sdk';
import { DemoWrapper } from '@/components/DemoWrapper/DemoWrapper';

export type HomeProps = { trainList: TrainType[] };

export const Home = ({ trainList }: HomeProps) => {
  return (
    <DemoWrapper>
      <main className="min-w-screen min-h-screen bg-slate-50 flex items-center justify-center">
        <article className="w-[calc(100vw-250px)] h-[calc(100vh-50px)]">
          <TrainScheduleDisplay>
            <div className="flex justify-between relative">
              <EmbeddedDisplay
                content={new Date().toISOString().split('T')[0]}
              />
              <h1 className="text-amber-300 text-5xl font-bold tracking-widest absolute top-7 left-[37%]">
                Train Schedule
              </h1>
              <EmbeddedDisplay content={<Clock />} />
            </div>

            <div className="text-amber-300 font-bold tracking-widest px-3 text-xl grid grid-cols-[1fr_250px_250px_150px_150px]">
              <div>Train Firm Name</div>
              <div>From</div>
              <div>To</div>
              <div>Arrival</div>
              <div>Departure</div>
            </div>

            <div className="w-full h-full overflow-y-scroll scrollbar-hidden relative">
              {trainList.map((train) => (
                <Train key={train.trainId} train={train} />
              ))}
            </div>
          </TrainScheduleDisplay>
        </article>
      </main>
    </DemoWrapper>
  );
};
