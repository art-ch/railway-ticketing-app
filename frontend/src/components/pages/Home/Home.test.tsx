import {
  BreakpointState,
  useBreakpoint
} from '@/hooks/useBreakpoint/useBreakpoint';
import { render } from '@testing-library/react';
import { Home } from './Home';
import { Train } from 'railway-ticketing-app-sdk';

jest.mock('@/hooks/useBreakpoint/useBreakpoint', () => ({
  useBreakpoint: jest.fn().mockReturnValue({ xl: true } as BreakpointState)
}));

jest.mock('./components/Clock/Clock', () => ({
  Clock: () => <div>Clock</div>
}));

jest.mock('@/utils/formatters', () => ({
  getFormattedTime: jest.fn().mockReturnValue('12:00')
}));

const useBreakpointMock = useBreakpoint as jest.MockedFunction<
  typeof useBreakpoint
>;

describe('Home component', () => {
  // Store the original Date
  const originalDate = global.Date;

  // Setup mock date before tests
  beforeAll(() => {
    // Mock the Date to return a fixed date
    const mockDate = new Date('2023-01-01T12:00:00Z');
    global.Date = class extends Date {
      constructor() {
        super();
        return mockDate;
      }

      static now() {
        return mockDate.getTime();
      }
    } as DateConstructor;
  });

  // Restore original Date after tests
  afterAll(() => {
    global.Date = originalDate;
  });

  const trainList = [
    { trainId: '1', seats: [{ seatNumber: 1, isBooked: false }] }
  ] as Train[];

  it('should render correctly', () => {
    const { container } = render(<Home trainList={trainList} />);

    expect(container).toMatchSnapshot();
  });

  it('should show demo notice for small screen', () => {
    useBreakpointMock.mockReturnValue({ xl: false } as BreakpointState);

    const { container } = render(<Home trainList={trainList} />);

    expect(container).toMatchSnapshot();
  });
});
