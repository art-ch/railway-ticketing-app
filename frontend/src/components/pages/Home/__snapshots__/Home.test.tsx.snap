// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render correctly 1`] = `
<div>
  <main
    class="min-w-screen min-h-screen bg-slate-50 flex items-center justify-center"
  >
    <article
      class="w-[calc(100vw-250px)] h-[calc(100vh-50px)]"
    >
      <div
        class="w-full h-full shadow-[10px_10px_80px_0px_grey] border-[15px] border-slate-200"
      >
        <div
          class="w-full h-full border-[3px] border-slate-400"
        >
          <div
            class="w-full h-full bg-zinc-900 flex flex-col gap-5"
          >
            <div
              class="flex justify-between relative"
            >
              <div
                class="border-[8px] border-slate-200"
              >
                <div
                  class="min-w-30 min-h-full border-[4px] border-slate-400"
                >
                  <div
                    class="text-white text-5xl p-3 font-dot-matrix font-bold"
                  >
                    2023-01-01
                  </div>
                </div>
              </div>
              <h1
                class="text-amber-300 text-5xl font-bold tracking-widest absolute top-7 left-[37%]"
              >
                Train Schedule
              </h1>
              <div
                class="border-[8px] border-slate-200"
              >
                <div
                  class="min-w-30 min-h-full border-[4px] border-slate-400"
                >
                  <div
                    class="text-white text-5xl p-3 font-dot-matrix font-bold"
                  >
                    <div>
                      Clock
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="text-amber-300 font-bold tracking-widest px-3 text-xl grid grid-cols-[1fr_250px_250px_150px_150px]"
            >
              <div>
                Train Firm Name
              </div>
              <div>
                From
              </div>
              <div>
                To
              </div>
              <div>
                Arrival
              </div>
              <div>
                Departure
              </div>
            </div>
            <div
              class="w-full h-full overflow-y-scroll scrollbar-hidden relative"
            >
              <a
                class="text-white text-3xl p-3 font-dot-matrix font-bold grid grid-cols-[1fr_250px_250px_150px_150px] hover:bg-amber-300 hover:text-black hover:font-extrabold hover:cursor-pointer"
                href="/1"
                target="blank"
              >
                <div />
                <div />
                <div />
                <div>
                  12:00
                </div>
                <div>
                  12:00
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </article>
  </main>
</div>
`;

exports[`Home component should show demo notice for small screen 1`] = `
<div>
  <div>
    This demo is intended to be shown on desktop
  </div>
</div>
`;
